core/charm/repository/charmhub.go

```diff
@@ -294,6 +334,45 @@ func (c *CharmHubRepository) retryResolveWithPreferredChannel(charmURL *charm.UR
 	}, nil
 }
 
+func transformRefreshResult(curl *charm.URL, refreshResult transport.RefreshResponse) (corecharm.EssentialMetadata, error) {
+	if refreshResult.Entity.MetadataYAML == "" {
+		return corecharm.EssentialMetadata{}, errors.Errorf("charmhub refresh response for %q does not include the contents of metadata.yaml", curl)
+	}
+	chMeta, err := charm.ReadMeta(strings.NewReader(refreshResult.Entity.MetadataYAML))
+	if err != nil {
+		return corecharm.EssentialMetadata{}, errors.Annotatef(err, "parsing metadata.yaml for %q", curl)
+	}
+
+	// It's okay for a charm not to have a config.yaml. But the CharmHub
+	// API currently returns "{}\n" for charms that have no config.yaml,
+	// so handle that as no config as well.
+	var chConfig *charm.Config
+	configYAML := refreshResult.Entity.ConfigYAML
+	if configYAML == "" || strings.TrimSpace(configYAML) == "{}" {
```

Comment:
I'm not a fan of this. I'd rather just unmarshal yaml and see if it's empty.

Reply:
This is a code move.. we do it in `cmd/juju/charmhub/convert.go` too.

Reply:
Pondering the issue, I'm taking out that code and just doing ReadConfig. It was related to juju/charmub alpha development things. ReadConfig handles the nil config as it should.